### PR TITLE
site: Add Contour v1.2.0 to the Kubernetes support matrix

### LINKED ISSUE #2251: update https://projectcontour.io/resources/kubernetes/ for 1.2.0 release
Please update https://projectcontour.io/resources/kubernetes/ to include the versions of k8s supported by Contour 1.2.0

### PR DESCRIPTION
**Pull request by stevesloka**
Fixes #2251 

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: site/_resources/kubernetes.md [hunk 1]
```diff
@@ -11,13 +11,14 @@ The `client-go` package includes a [compatibility matrix][2] as to what Kubernet
...
-| 1.17.x | Supported | Supported | Supported | 
-| 1.18.x | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup> | Not Supported <sup>1</sup> |
+| Kubernetes version | Contour v1.0.0 | Contour v1.0.1 | Contour v1.1.0 | Contour v1.2.0 |
+| ------------ | :-----------: | :-----------: | :----------: | :--------: |
+| 1.15.x | Supported | Supported | Supported | Supported |
+| 1.16.x | Supported | Supported | Supported | Supported |
+| 1.17.x | Supported | Supported | Supported | Supported |
+| 1.18.x | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup> | Not Supported <sup>1</sup> | Not Supported <sup>1</sup> |
 
+<br>
```
Comment (davecheney) [inline site/_resources/kubernetes.md]: please delete this, mixing html and markdown makes Jekyll sad.

**Comment by davecheney** [comment]
Assigning @youngnick as the reviewer

File: site/_resources/kubernetes.md (hunk 1 above)
Comment (stevesloka) [reply site/_resources/kubernetes.md]: It works in Netlify just fine: https://deploy-preview-2263--projectcontour.netlify.com/resources/kubernetes/

Where does the sadness come from?

**Comment by stevesloka** [comment]
> We upgraded client-go in contour 1.2, but the set of supported k8s versions did not change, why is that?

I guess you are correct, let me update again. This page got created in the 1.2 cycle and I mixed up the versioned docs vs the unversioned.

**Comment by stevesloka** [comment]
@davecheney @youngnick I updated the Matrix. Thanks for spotting my mistake @davecheney. 

Contour v1.0 & v1.1 used client-go v0.15.0, Contour v1.2 uses client-go v0.17.

File: site/_resources/kubernetes.md (hunk 1 above)
Comment (davecheney) [reply site/_resources/kubernetes.md]: See #2234 #2252

Comment (davecheney) [reply site/_resources/kubernetes.md]: Also `<br>` adds a line break inside html, did you mean `<hr>`?

File: site/Gemfile.lock [hunk 2]
```diff
@@ -122,4 +122,4 @@ DEPENDENCIES
   jekyll-titles-from-headings (~> 0.5.3)
 
 BUNDLED WITH
-   2.0.2
```
Comment (davecheney) [inline site/Gemfile.lock]: please revert this noop change

File: site/_resources/kubernetes.md [hunk 3]
```diff
@@ -11,13 +11,16 @@ The `client-go` package includes a [compatibility matrix][2] as to what Kubernet
...
-| 1.17.x | Supported | Supported | Supported | 
-| 1.18.x | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup> | Not Supported <sup>1</sup> |
+| Kubernetes version | Contour v1.0.0 | Contour v1.0.1 | Contour v1.1.0 | Contour v1.2.0 |
+| ------------ | :-----------: | :-----------: | :----------: | :--------: |
+| 1.13.x | Supported | Supported | Supported | Not Supported<sup>1</sup> |
+| 1.14.x | Supported | Supported | Supported | Not Supported<sup>1</sup> |
+| 1.15.x | Supported | Supported | Supported | Supported |
+| 1.16.x | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup> | Supported |
+| 1.17.x | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup> | Supported |
+| 1.18.x | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup> | Not Supported <sup>1</sup> | Not Supported <sup>1</sup> |
```
Comment (davecheney) [inline site/_resources/kubernetes.md]: I'd prefer that we don't list versions that don't exist yet, then we don't have to say if they are supported or not. It's a Heisenberg thing.

File: site/_resources/kubernetes.md (hunk 1 above)
Comment (stevesloka) [reply site/_resources/kubernetes.md]: I just wanted to add some space after the table. I'll delete the `<br>` but it looks fine in Netlify and my local build.

**Pull request merged**